src/Compilers/Shared/BuildServerConnection.cs

```diff
@@ -332,7 +336,7 @@ internal sealed class BuildServerConnection
                 {
                     // It is okay for this call to fail.  Errors will be reflected in the
                     // IsConnected property which will be read on the next iteration of the
-                    logger.LogException(e, $"Error poking pipe {identifier}.");
+                    logger.LogException(e, $"Error poking pipe {requestId}.");
```

Comment:
nit: would it be worthwhile to have the logger handle the requestId (add it to the log) so that we don't have to pass it around in so many places that already have the logger? Or do the requestId and logger have different lifetimes?

Reply:
Mm... I see that for VBCSCompiler we don't create the requestId at the same time that we create the logger, so my proposal probably wouldn't work.

---
In reply to: [612732135](https://github.com/dotnet/roslyn/pull/52572#discussion_r612732135) [](ancestors = 612732135)

Reply:
That wouldn't work though. The intent of the request id is to track a compilation request across processes. The request originates in MSBuild, is passed to VBCSCompiler, then back to MSBuild once the request completes. There is no way to generate this before hand on both sides of the pipe.